src/random_ray/iteration.cpp

```diff
@@ -0,0 +1,610 @@
+
+    // Execute all tallying tasks, if this is an active batch
+    if (simulation::current_batch > settings::n_inactive && mpi::master) {
+
+      // Generate mapping between source regions and tallies
+      if (!mapped_all_tallies) {
+        mapped_all_tallies = convert_source_regions_to_tallies();
+      }
+
+      // Use above mapping to contribute FSR flux data to appropriate tallies
+      random_ray_tally();
+
+// Add this iteration's scalar flux estimate to final accumulated estimate
+#pragma omp parallel for
+      for (int64_t se = 0; se < random_ray::n_source_elements; se++) {
+        random_ray::scalar_flux_final[se] += random_ray::scalar_flux_new[se];
+      }
+    }
+
+    // Set phi_old = phi_new
+    random_ray::scalar_flux_old.swap(random_ray::scalar_flux_new);
+
+    // Check for any obvious insabilities/nans/infs
+    instability_check(n_hits, k_eff, avg_miss_rate);
+
+    // Finalize the current batch
+    finalize_generation();
+    finalize_batch();
+  } // End random ray power iteration loop
+
+  openmc::simulation::time_total.stop();
+
+  // Finalize OpenMC
+  openmc_simulation_finalize();
+
+  // Reduce number of intersections
+#ifdef OPENMC_MPI
+  if (mpi::n_procs > 1) {
+    uint64_t total_geometric_intersections_reduced = 0;
+    MPI_Reduce(&total_geometric_intersections,
+      &total_geometric_intersections_reduced, 1, MPI_UNSIGNED_LONG, MPI_SUM, 0,
+      mpi::intracomm);
+    total_geometric_intersections = total_geometric_intersections_reduced;
+  }
+#endif
+
+  // Print random ray results
+  if (mpi::master) {
+    print_results_random_ray(
+      total_geometric_intersections, avg_miss_rate / settings::n_batches);
+    if (model::plots.size() > 0) {
+      plot_3D_vtk();
+    }
+  }
+
+  return 0;
+}
+
+// Compute new estimate of scattering + fission sources in each source region
+// based on the flux estimate from the previous iteration.
+void update_neutron_source(double k_eff)
+{
+  simulation::time_update_src.start();
+
+  double inverse_k_eff = 1.0 / k_eff;
+  int negroups = data::mg.num_energy_groups_;
+
+  // Temperature and angle indices, if using multiple temperature
+  // data sets and/or anisotropic data sets.
+  // TODO: Currently assumes we are only using single temp/single
+  // angle data.
+  const int t = 0;
+  const int a = 0;
+
+#pragma omp parallel for
+  for (int sr = 0; sr < random_ray::n_source_regions; sr++) {
+    int material = random_ray::material[sr];
+
+    for (int energy_group_out = 0; energy_group_out < negroups;
+         energy_group_out++) {
+      float Sigma_t = data::mg.macro_xs_[material].get_xs(
+        MgxsType::TOTAL, energy_group_out, nullptr, nullptr, nullptr, t, a);
+      float scatter_source = 0.0f;
+      float fission_source = 0.0f;
+
+      for (int energy_group_in = 0; energy_group_in < negroups;
+           energy_group_in++) {
+        float scalar_flux =
+          random_ray::scalar_flux_old[sr * negroups + energy_group_in];
+        float Sigma_s =
```

Comment:
I believe the convention is to not capitalize variables except in cases like "E", and IIRC these are not in the exception list. Same for "Chi" below.

Reply:
Good thinking -- I can make that change.